packages/venia-ui/lib/components/SearchBar/__tests__/searchField.spec.js

```diff
@@ -1,39 +1,53 @@
 import React from 'react';
+import { useLocation } from 'react-router-dom';
 import { Form } from 'informed';
 import { act } from 'react-test-renderer';
 import { createTestInstance } from '@magento/peregrine';
-
 import Trigger from '../../Trigger';
 import SearchField from '../searchField';
 
+jest.mock('react-router-dom', () => {
+    return {
+        useLocation: jest.fn()
+    };
+});
+
+jest.mock('@magento/peregrine/lib/context/app', () => {
+    const state = {
+        searchOpen: true
+    };
+    const api = {};
+    return {
+        useAppContext: jest.fn(() => [state, api])
+    };
+});
+
 jest.mock('../../../classify');
 jest.mock('../../Trigger', () => () => null);
 
 const onChange = jest.fn();
 const onFocus = jest.fn();
 
 test('renders correctly', () => {
+    const location = { pathname: '/' };
+    useLocation.mockReturnValueOnce(location);
```

Comment:
Instead of writing this in every test, can we move this to `beforeEach` and change it in a certain test if needed? Or like you did for `useLocation` in `packages/peregrine/lib/hooks/__tests__/useSearchParam.spec.js `.

Reply:
Sure, I could just have it return this by default and only do the override in the one test.